### PR TITLE
If the version file is empty, complain.

### PR DESCRIPTION
**Pull request by zmc**
We *could* work around this, but I don't want to mask bugs in the build
system.

Signed-off-by: Zack Cerza <zack@redhat.com>

### CODE REVIEW

**Comment by andrewschoen** [comment]
Agreed that masking bugs in the build system is something we don't want to do, but also failing jobs is a pain too.

Is there a way to make it complain loud enough that we'd see, but allow the job to continue and possibly finish by giving need_install the sha1 instead of version?

File: teuthology/task/kernel.py [hunk 1]
```diff
@@ -1111,6 +1113,11 @@ def task(ctx, config):
                     log.debug('failed to get utsrelease string using url {url}'.format(
                               url=version_url))
 
+                if not version:
+                    raise VersionNotFoundError(
+                        "{url} is invalid; contains {ver}".format(
```
Comment (andrewschoen) [inline teuthology/task/kernel.py]: considering that version is most likely an empty string, does it make sense to print it?

**Comment by zmc** [comment]
Hmm, yeah, it'd be just as easy to log a warning. Let me think about that for a minute...

**Comment by andrewschoen** [comment]
My only concern with the warning message is if the job passes with the sha1 instead of the version, we never see the warning message.  I surely don't make a habit of reading logs for jobs that pass.

**Comment by zmc** [comment]
Yeah, exactly. I kind of want to force us to deal with build system issues like this. I'll bring it up in standup and see if we can get a consensus.

**Pull request merged**